### PR TITLE
Extend (legacy)meta server configuration.

### PR DESCRIPTION
**Pull request by JaredTan95**
Support to override meta server URL through environment variables.

### CODE REVIEW

**Comment by nobodyiam** [comment]
Thanks!

Normally, we think system properties have higher priority than system environment properties, is it ok to adjust the logic?

Also, after adding system environment logic, the code looks a little complex, is it better to extract the get meta data logic as a standalone method?

**Comment by JaredTan95** [comment]
ok,I'll do some refactoring.

File: apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java [hunk 1]
```diff
@@ -24,18 +27,28 @@ private void initialize() {
...
+
+    domains.put(Env.LOCAL, getMeteServerConf(env, prop, "local_meta", "local.meta"));
+    domains.put(Env.DEV, getMeteServerConf(env, prop, "dev_meta", "dev.meta"));
+    domains.put(Env.FAT, getMeteServerConf(env, prop, "fat_meta", "fat.meta"));
+    domains.put(Env.UAT, getMeteServerConf(env, prop, "uat_meta", "uat.meta"));
+    domains.put(Env.LPT, getMeteServerConf(env, prop, "lpt_meta", "lpt.meta"));
+    domains.put(Env.PRO, getMeteServerConf(env, prop, "pro_meta", "pro.meta"));
+  }
+
+  public String getMeteServerConf(Properties env, Properties prop, String sourceName,
```
Comment (nobodyiam) [inline apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java]: getMeteServerConf -> getMetaServerAddress

File: apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java [hunk 2]
```diff
@@ -24,18 +27,28 @@ private void initialize() {
...
+      String propName) {
+    // 1. Get from System Property.
+    String metaAddress = env.getProperty(sourceName);
```
Comment (nobodyiam) [inline apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java]: how about System.getProperty(sourceName)?

File: apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java [hunk 3]
```diff
@@ -24,18 +27,28 @@ private void initialize() {
...
+    if (Strings.isNullOrEmpty(metaAddress)) {
+      // 2. Get from OS environment variable, which could not contain dot and is normally in UPPER case,like DEV_META.
+      metaAddress = getenv(sourceName.toUpperCase());
```
Comment (nobodyiam) [inline apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java]: System.getenv might be more clear :-)

Comment (JaredTan95) [reply apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java]: tks，😄

**Comment by nobodyiam** [comment]
Looks good to me, would you please stash the commits to one commit?

**Comment by JaredTan95** [comment]
https://github.com/ctripcorp/apollo/pull/1565

**Pull request closed without merging**